src/mlpack/methods/ann/activation_functions/isrlu_function.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file isrlu_function.hpp
+ * @author Pranav Reddy
+ *
+ * Definition and implementation of the ISRLU function
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ISRLU_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ISRLU_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace ann /** Artifical Neural Network. */{
+
+/**
+ * The ISRU function, defined by
+ * 
+ * @f{eqnarray*}{
+ * f(x) &=& x/sqrt(1 + alpha*x*x) if x<0
+ *      &=& x                     if x>=0
+ *
+ * f'(x) &=& cube(1/sqrt(1 + alpha*x*x)) if x<0
+ *       &=& 1                           if x>=0
+ * @f
+ */
+
+class ISRLUFunction
+{
+
+ public:
+  /**
+   * Computes the ISRLU function
+   * 
+   * @param x Input data.
+   * @param alpha paramter, default value = 0.1
+   * @return f(x)
+   */
+  static double Fn(const double x, const double alpha = 0.1)
```

Comment:
Can you implement this as a layer, similar to: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/leaky_relu.hpp, because this function is most likely used in combination with the base layer, and there is no simple way to use a different alpha value.